quickstart/201-web-app-on-docker-container/main.tf

```diff
@@ -0,0 +1,80 @@
+resource "azurerm_resource_group" "default" {
+  name     = "${var.name_prefix}-rg"
+  location = var.location
+}
+
+resource "azurerm_user_assigned_identity" "default" {
+  name                = "${var.name_prefix}-uai"
+  resource_group_name = azurerm_resource_group.default.name
+  location            = azurerm_resource_group.default.location
+}
+
+resource "azurerm_container_registry" "default" {
+  name                = "${var.name_prefix}acr"
+  resource_group_name = azurerm_resource_group.default.name
+  location            = azurerm_resource_group.default.location
+  sku                 = "Premium"
+  identity {
+    type = "UserAssigned"
+    identity_ids = [
+      azurerm_user_assigned_identity.default.id
+    ]
+  }
+}
+
+resource "azurerm_service_plan" "default_linux" {
+  name                = "${var.name_prefix}-sp-linux"
+  location            = azurerm_resource_group.default.location
+  resource_group_name = azurerm_resource_group.default.name
+  sku_name            = "P1v3"
+  os_type             = "Linux"
+}
+
+resource "azurerm_service_plan" "default_windows" {
+  name                = "${var.name_prefix}-sp-windows"
+  location            = azurerm_resource_group.default.location
+  resource_group_name = azurerm_resource_group.default.name
+  sku_name            = "P1v3"
+  os_type             = "Windows"
+}
+
+# when setting docker container for web app, please use app_setting block to specify the container registry information such as URL and credentials, instead of docker_container_registry property in application_stack block. Service will try to locate the docker image based on linuxFxVersion/ windowsFxVersion property and they are composed following the format: DOCKER|containerRegistry/containerName:containerTag in terraform provider.
+resource "azurerm_linux_web_app" "default" {
+  name                = "${var.name_prefix}-lwa"
+  location            = azurerm_resource_group.default.location
+  resource_group_name = azurerm_resource_group.default.name
+  service_plan_id     = azurerm_service_plan.default_linux.id
+  app_settings = {
+    "DOCKER_REGISTRY_SERVER_URL"          = "https://${azurerm_container_registry.default.name}.azurecr.io"
+    "DOCKER_REGISTRY_SERVER_USERNAME"     = ""
```

Comment:
I have one question for this empty `DOCKER_REGISTRY_SERVER_USERNAME` and `DOCKER_REGISTRY_SERVER_PASSWORD`, according to `azurerm_container_registry`'s [document](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/container_registry#anonymous_pull_enabled), the `anonymous_pull_enabled` is `false` by default so unauthenticated user would be blocked. Have we tested such configuration could work?

Reply:
I added the username and password information to login the ACR as its one way of accessing the private container registry, the other is use MSI.